lib/Models/CkanCatalogGroup.js

```diff
@@ -597,25 +620,41 @@ function populateGroupFromResults(ckanGroup, json) {
                         groupToAdd.id = groupId;
                     }
 
-                    addItem(resource, ckanGroup, item, extras, groupToAdd);
+                    addedItem = addItem(resource, ckanGroup, item, extras, groupToAdd);
 
                     if (!updating && groupToAdd.items.length) {
                         ckanGroup.add(groupToAdd);
                     }
                 }
             } else {
                 if (!ckanGroup.ungroupedTitle) {
-                    addItem(resource, ckanGroup, item, extras, ckanGroup);
+                    addedItem = addItem(resource, ckanGroup, item, extras, ckanGroup);
                 } else {
                     if (!defined(ungrouped)) {
                         ungrouped = new CatalogGroup(ckanGroup.terria);
                         ungrouped.name = ckanGroup.ungroupedTitle;
                         ungrouped.id = ckanGroup.uniqueId + '/_ungrouped';
                         ckanGroup.add(ungrouped);
                     }
-                    addItem(resource, ckanGroup, item, extras, ungrouped);
+                    addedItem = addItem(resource, ckanGroup, item, extras, ungrouped);
                 }
             }
+
+            if (defined(addedItem)) {
+                resourceItems.push(addedItem);
+            }
+        }
+
+        // If there's more than one resource item, and we're not using the resource name to name
+        // our items, then they'll all have the same name.  Add the type to the name to help
+        // distinguish them.
+        if (resourceItems.length > 1 && !ckanGroup.useResourceName) {
+            resourceItems.forEach(function(item) {
+                var typeName = CkanCatalogItem.shortHumanReadableTypeNames[item.type] || 'Other';
+                if (defined(typeName)) {
```

Comment:
Oops, that `if` check isn't needed at all anymore, since I added the `|| 'Other'`.  I'll remove it.